src/Microsoft.DotNet.Darc/DarcLib/Remote.cs

```diff
@@ -188,11 +189,22 @@ public async Task<List<GitFile>> CommitUpdatesAsync(
 
         SemanticVersion targetDotNetVersion = null;
         var mayNeedArcadeUpdate = arcadeItem != null && repoUri != arcadeItem.RepoUri;
+        // If we find version files in src/arcade, we know we're working with a VMR
+        bool repoIsVmr = true;
 
         if (mayNeedArcadeUpdate)
         {
             IDependencyFileManager arcadeFileManager = await remoteFactory.CreateDependencyFileManagerAsync(arcadeItem.RepoUri);
-            targetDotNetVersion = await arcadeFileManager.ReadToolsDotnetVersionAsync(arcadeItem.RepoUri, arcadeItem.Commit);
+            try
+            {
+                targetDotNetVersion = await arcadeFileManager.ReadToolsDotnetVersionAsync(arcadeItem.RepoUri, arcadeItem.Commit, repoIsVmr);
+            }
+            catch (DependencyFileNotFoundException)
+            {
+                // global.json not found in src/arcade meaning that repo is not the VMR
+                repoIsVmr = false;
+                targetDotNetVersion = await arcadeFileManager.ReadToolsDotnetVersionAsync(arcadeItem.RepoUri, arcadeItem.Commit, repoIsVmr);
+            }
```

Comment:
Maybe we could just pass the flag from outside so we don't need to do this?

We can tell this in the higher levels more easy because we have other metadata at hand and we don't need to fire a request for this (maybe it even does retries?).

Reply:
But which metadata would be able to tell us about this? This code runs only during normal Dependency Flow updates, and is needed when we're flowing arcade our of the VMR

Reply:
So the dependency flow code knows if a subscription is a back/forward flow from the subscription metadata

Reply:
But this class is used in normal non code flow subscriptions. so if we have a `VMR` -> `product repo PR` we don't have that sort of metadata

Reply:
Aah, right. Got it, we won't have that information.